src/session.ts

```diff
@@ -188,6 +199,135 @@ export function valueSession(): void {
+  }
+
+  // Start tracking items if at least one of these is true
+  // 1) We have run at least 100 barf turns
+  // 2) We have less than 200 adv to run
+  if (
+    !marginalSession &&
+    (totalTurnsPlayed() - barfSessionStartTurns >= 100 ||
+      (myAdventures() <= 200 + 25 + globalOptions.saveTurns &&
+        myAdventures() > globalOptions.saveTurns + 25))
+  ) {
+    marginalSession = Session.current();
+    numTrackedItemTurns = myAdventures() - globalOptions.saveTurns - 25;
+  }
+
+  // Start tracking meat if we have less than 75 turns left
+  // Also create a backup tracker for items
+
+  if (
+    (!get("_garboMarginalMeatCheckpoint") || !get("_garboMarginalMeatTurns")) &&
+    myAdventures() - 25 - globalOptions.saveTurns <= 50 &&
+    myAdventures() > 25 + globalOptions.saveTurns
+  ) {
+    const { meat, items } = sessionSinceStart().value(garboValue);
+    const numTrackedMeatTurns = myAdventures() - 25 - globalOptions.saveTurns;
+    setProperty("_garboMarginalMeatCheckpoint", meat.toFixed(0));
+    setProperty("_garboMarginalItemCheckpoint", items.toFixed(0));
+    setProperty("_garboMarginalMeatTurns", numTrackedMeatTurns.toFixed(0));
+  }
+
+  // Stop tracking meat if we have less than 25 turns left
+  if (
+    get("_garboMarginalMeatCheckpoint") &&
+    get("_garboMarginalMeatTurns") &&
+    continueMeatTracking &&
+    myAdventures() - 25 - globalOptions.saveTurns <= 0
+  ) {
+    continueMeatTracking = false;
+    const { meat, items } = sessionSinceStart().value(garboValue);
+    const meatDiff = meat - get("_garboMarginalMeatCheckpoint", 0);
+    const itemDiff = items - get("_garboMarginalItemCheckpoint", 0);
+    setProperty("_garboMarginalMeatValue", meatDiff.toFixed(0));
+    setProperty("_garboMarginalItemValue", itemDiff.toFixed(0));
+    if (marginalSession) marginalSessionDiff = Session.current().diff(marginalSession);
+  }
+}
+
+function printMarginalSession(): void {
+  if (myInebriety() > inebrietyLimit() || myAdventures() > globalOptions.saveTurns) return;
+
+  if (get("_garboMarginalMeatValue") && get("_garboMarginalMeatTurns")) {
+    const meat = get("_garboMarginalMeatValue", 0);
+    const meatTurns = get("_garboMarginalMeatTurns", 0);
+
+    if (meatTurns <= 0) {
+      print("Error in estimating marginal MPA - meat turns tracked = 0", "red");
+      return;
+    }
+
+    const MPA = meat / meatTurns;
+
+    // Only evaluate item outliers if we have run a good number of turns (to reduce variance)
+    if (
+      marginalSessionDiff &&
+      barfSession &&
+      numTrackedItemTurns &&
+      numTrackedItemTurns >= Math.max(50, meatTurns)
+    ) {
+      const { items, itemDetails } = marginalSessionDiff.value(garboValue);
+      const barfItemDetails = Session.current().diff(barfSession).value(garboValue).itemDetails;
+      const outlierItemDetails = itemDetails
+        .filter(
+          (detail) =>
+            detail.quantity === 1 &&
+            detail.value >= 1000 &&
```

Comment:
is 1000 a reasonable limit in assessing outlier items? that seems... low, to me, if we are talking true outliers. for instance, most candies dropped by robort will be over that value. many things from cleaver. knob visors. (ok the last one is a joke but still.) this feels like it will drag way too much of the profit into the outliers but not truly communicate why. what do yall think?

Reply:
These are the "outliers" on the breakfast leg that I got yesterday
> 1 fake hand worth 50000 total
> 1 Sourfinger worth 43650 total
> 1 Pok&eacute;-Gro fertilizer worth 13860 total
> 1 recording of Chorale of Companionship worth 9225 total
> 1 chocolate seal-clubbing club worth 7416 total
> 1 chocolate pasta spoon worth 7076 total
> 1 salt wages worth 6919 total
> 1 goblin water worth 6296 total
> 1 peppermint sprout worth 5992 total
> 1 chocolate disco ball worth 5396 total
> 1 chocolate stolen accordion worth 5265 total
> 1 Friendliness Beverage worth 4946 total
> 1 fudge-shaped hole in space-time worth 3600 total
> 1 Mer-kin fastjuice worth 2699 total
> 1 eagle feather worth 2520 total
> 1 Mer-kin strongjuice worth 2156 total
> 1 gummi salamander worth 2052 total
> 1 half-orchid worth 1566 total
> 1 Comet Drop worth 1526 total
> 1 sterno-flavored Hob-O worth 1522 total
> 1 red drunki-bear worth 1521 total
> 1 sugar shorts worth 1521 total
> 1 tiny dancer worth 1494 total
> 1 lead necklace worth 1492 total
> 1 wind-up Whatsian robot worth 1229 total
> 1 skin oil worth 1080 total
> 1 green drunki-bear worth 1063 total
> 1 8-bit banana worth 1053 total
> 1 mother's necklace worth 1000 total
> Marginal MPA: 5,090.3 [raw] + 1,704.26 [items] (2,314.05 [outliers]) = 6,794.56 [total] (7,404.35 [w/ outliers])
I suppose ut does grab a few things from cleaver, robort and screeges, and I'm comfortable raising that limit to 5000. But the 1000-5000 items (totalling 34040 meat) really do not skew the outlier MPA calculations as much as the heavy hitters (totalling 161095 meat for the > 5000 ones)